### DISCUSSION TITLE
Kitty Graphics Visual Artifacts

### DISCUSSION THREAD

**Discussion by AbeEstrada** [ghostty-org/ghostty#3803] [Feature Requests, Ideas] [answered]
![kittygfx1](https://github.com/user-attachments/assets/d2097dd8-4709-4d64-901a-1d4fa86b1b7f)

Visual artifacts appear in Ghostty during the rendering process of an image

Image: https://cdn.vox-cdn.com/uploads/chorus_image/image/73807500/Icebreaker_keyboard_desk.0.jpg
Left: Ghostty 2a4e5e1c
Right: Kitty 0.38.1
macOS 15.2

**Comment by jcollie** [comment]
I don't have any idea why it would be happening, but do you have `minimum-contrast` set? That might explain why it only affects darker parts of the image, but I wouldn't have thought that `minimum-contrast` affected Kitty images.

**Reply by jcollie** [reply]
I'm unable to reproduce, even with `minimum-contrast` set. I'm on Linux so this could be a macOS-only thing.

**Reply by AbeEstrada** [reply]
I don't have `minimum-contrast` set, I have `window-colorspace = display-p3` but it doesn't affect if I use it or not

**Reply by AbeEstrada** [reply, marked as answer]
It was the `custom-shader`
